src/tests/quantuminspire/projectq/test_backend_qx.py

```diff
@@ -358,21 +401,299 @@ def test_receive_multiple_flush(self, function_mock):
         self.assertEqual(backend.qasm, "")
         self.assertTrue(backend._clear)
 
+    @patch('quantuminspire.projectq.backend_qx.get_control_count')
+    def test_flush_with_no_measurements_but_nfsp(self, function_mock):
+        function_mock.return_value = 1
+        command_alloc0 = MagicMock(gate=Allocate, qubits=[[MagicMock(id=0)]])
+        command_alloc1 = MagicMock(gate=Allocate, qubits=[[MagicMock(id=1)]])
+        command_alloc2 = MagicMock(gate=Allocate, qubits=[[MagicMock(id=1)]])
+        command_dealloc1 = MagicMock(gate=Deallocate, qubits=[[MagicMock(id=1)]])
+        command = MagicMock(gate=NOT, qubits=[[MagicMock(id=0)]], control_qubits=[MagicMock(id=1)])
+        command_list = [command_alloc0, command_alloc1, command_dealloc1, command_alloc2, command,
+                        MagicMock(gate=FlushGate())]
+        api = MockApiClient()
+        backend = QIBackend(quantum_inspire_api=api, verbose=1)
+        backend.main_engine = MagicMock()
+        with patch('sys.stdout', new_callable=io.StringIO):
+            backend.receive(command_list)
+        self.assertEqual(backend.qasm, "")
+        self.assertTrue(backend._clear)
```

Comment:
hmmm, peaking into the box

Reply:
Restructured the unit tests to make valid use of protected members and not directly. Changed it for all the tests.